source/server/listener_manager_impl.cc

```diff
@@ -180,20 +205,33 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
   if (config.has_freebind()) {
     addListenSocketOptions(Network::SocketOptionFactory::buildIpFreebindOptions());
   }
-  if (config.has_tcp_fast_open_queue_length()) {
-    addListenSocketOptions(Network::SocketOptionFactory::buildTcpFastOpenOptions(
-        config.tcp_fast_open_queue_length().value()));
-  }
-
   if (!config.socket_options().empty()) {
     addListenSocketOptions(
         Network::SocketOptionFactory::buildLiteralOptions(config.socket_options()));
   }
 
   if (!config.listener_filters().empty()) {
-    listener_filter_factories_ =
-        parent_.factory_.createListenerFilterFactoryList(config.listener_filters(), *this);
+    switch (socket_type_) {
+    case Network::Address::SocketType::Datagram:
+      udp_listener_filter_factories_ =
+          parent_.factory_.createUdpListenerFilterFactoryList(config.listener_filters(), *this);
+      // TODO(sumukhs): Determine if returning here is the right thing to do as udp listeners are
```

Comment:
Should we return here by silently ignoring other unrelated fields that might be present?? (like filter chains etc?)

Reply:
I think it's fine, or you could warn if the fields are set I suppose. @danzh2010 WDYT?

Reply:
I am fine either ways.. I will leave it as it is now and quic can modify this logic when there are more changes here.